### PR TITLE
Remove `this` from `VGridSkeleton` props `default`

### LINKED ISSUE #3268: Remove `this` from `VGridSkeleton` props `default`
## Problem

`VGridSkeleton` uses `this` in the props' `default` function. This causes a build error in Vue 3.
https://github.com/WordPress/openverse/blob/964c5cca996ad0073318a01f4637df64956cd7e1/frontend/src/components/VSkeleton/VGridSkeleton.vue#L45-L52

## Description

The setting of the default element count should be moved to the `setup` function:

```
const elementCount = computed(() => {
  if (props.numElems) return props.numElems
   if (props.isForTab === "all") return 20
   if (props.isForTab === "image") return 30
   return 8
})
```

The `numElems` would need to be replaced with `elementCount` in the `template`, too.

### PR DESCRIPTION
**Pull request by Ashhar-24**
## Fixes

Fixes #3268 by @Ashhar-24

## Description

As  mentioned in the issue #3268 that `VGridSkeleton` uses `this` in the props' `default` function and this causes a build error in Vue 3.
So I have moved default element count to the `setup` function and the `numElems` is replaced with `elementCount` in the `template`.

Note: if this PR is ready after #3322 is merged, this PR needs to remove the TODO from `packages/eslint-plugin/src/configs/vue.ts`:
```
    // TODO: disable after https://github.com/WordPress/openverse/issues/3268 is fixed
    "vue/no-deprecated-props-default-this": "off",
```

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VSkeleton/VGridSkeleton.vue [hunk 1]
```diff
@@ -44,19 +44,22 @@ export default defineComponent({
...
       type: Number,
-      default: function () {
-        if (this.isForTab === "all") return 20
-        if (this.isForTab === "image") return 30
-        return 8
-      },
     },
   },
-  setup() {
+  setup(props: { numElems: any; isForTab: string; }) {
```
Comment (obulat) [inline frontend/src/components/VSkeleton/VGridSkeleton.vue:49]:
```suggestion
  setup(props) {
```
The types of props are defined above, and we don't need to add any type information here.

**Review by obulat — CHANGES_REQUESTED** [review]
Awesome work, @Ashhar-24 !
I think the only thing that's preventing the linting step in the CI is the `any` in the `setup` call. I added a suggestion how to fix it. If the CI passes after that fix, I'll approved this PR. Otherwise I'll level another review :)

File: frontend/src/components/VSkeleton/VGridSkeleton.vue (hunk 1 above)
Comment (Ashhar-24) [reply frontend/src/components/VSkeleton/VGridSkeleton.vue]: Made the suggested changes.
But still the CI test is failing @obulat

**Comment by obulat** [comment]
You will need to run `just lint` and commit the changes.

**Comment by Ashhar-24** [comment]
> You will need to run `just lint` and commit the changes.

image
This is error message that i'm receiveing.

File: frontend/src/components/VSkeleton/VGridSkeleton.vue [hunk 2]
```diff
@@ -44,19 +44,22 @@ export default defineComponent({
...
     }
 
-    return { getRandomSize }
+    // Calculate the default element count based on isForTab
+    const elementCount = computed(() => {
+          if (props.numElems) return props.numElems;
+          if (props.isForTab === "all") return 20;
+          if (props.isForTab === "image") return 30;
+          return 8;
+        });
```
Comment (obulat) [inline frontend/src/components/VSkeleton/VGridSkeleton.vue:60]:
```suggestion
      if (props.numElems) return props.numElems
      if (props.isForTab === "all") return 20
      if (props.isForTab === "image") return 30
      return 8
    })
```

This is the change that the linter makes. The CI should pass if you accept it, @Ashhar-24 .
Did you run `just install` or `just node-install` before running the `just lint` command? It seems that the reason you are getting the error locally is the fact that not all dependencies are installed.

**Review by obulat — CHANGES_REQUESTED** [review]

File: frontend/src/components/VSkeleton/VGridSkeleton.vue [hunk 3]
```diff
@@ -44,19 +44,21 @@ export default defineComponent({
...
     },
   },
-  setup() {
+  setup(props) {
     function getRandomSize(max = 300, min = 100) {
       return Math.floor(Math.random() * (max - min) + min)
     }
 
-    return { getRandomSize }
+    // Calculate the default element count based on isForTab
```
Comment (obulat) [inline frontend/src/components/VSkeleton/VGridSkeleton.vue:54]:
```suggestion
    const elementCount = computed(() => {
    // Calculate the default element count based on isForTab
```

File: frontend/src/components/VSkeleton/VGridSkeleton.vue [hunk 4]
```diff
@@ -44,19 +44,22 @@ export default defineComponent({
...
   },
-  setup() {
+  setup(props) {
     function getRandomSize(max = 300, min = 100) {
       return Math.floor(Math.random() * (max - min) + min)
     }
 
-    return { getRandomSize }
+    const elementCount = computed(() => {
+    // Calculate the default element count based on isForTab
```
Comment (obulat) [inline frontend/src/components/VSkeleton/VGridSkeleton.vue:55]:
```suggestion
      // Calculate the default element count based on isForTab
```

**Review by obulat — APPROVED** [review]
I went ahead and committed the suggested changes, @Ashhar-24. Now it's ready to go 🚀

**Review by dhruvkb — APPROVED** [review]
LGTM! Thanks for the contribution @Ashhar-24!

**Comment by Ashhar-24** [comment]
Thank you @obulat for your constant support.
Was nice working with the issue.

**Pull request merged**